File: client/src/types/account.ts

export type SignupCredentials = {
    fullname: string,
    address: string,

    email: string,
    password: string
    role:string
}

export type LoginCredentials = {
    device: string
    email: string,
    password: string
}

export interface InputData {
    label: keyof SignupCredentials; 
    type: string;
    placeholder: string;
    minLength?: number;
    pattern?: RegExp;
  
}

export const validEmail = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;


export type PersonnelAccounts = {
    user_id: number,
    full_name: string,
    email: string,
    address: string,
    password?: string,
}


export type adminLoginCredentials = {
    email: string,
    password: string
}

export interface InputType {
    label: keyof adminLoginCredentials; 
    type: string;
    placeholder: string;
}


export type AdminData = {
    id: number,
    email: string
}


export type PersonnelAccountEdit = {
    id: number,
    full_name: string,
    address: string,
    email: string,
    password: string,
}

File: types/account.go
package types

import (
	"github.com/golang-jwt/jwt/v5"
)

type ContextKey string

const AdminCtxKey ContextKey = "adminPayload"


type SignupCredentials struct {
	FullName string `json:"fullname"`
	Address  string `json:"address"`
	Email    string `json:"email"`
	Password string `json:"password"`
	Role     string `json:"role"`
}

type LoginCredentials struct {
	Email    string `json:"email"`
	Password string `json:"password"`
}

type AdminData struct {
	ID       int64  `json:"id"`
	Email    string `json:"email"`
	Password string `json:"password"`
}

type AdminLoginData struct{
	ID       int64  `json:"id"`
	Email    string `json:"email"`
}

type UserInfo struct {
	ID       int64
	FullName string
	Address  string
	Email    string
	Password string
	Role     string
}

type SuccessLogin struct {
	UserID		 int64 	`json:"user_id"`
	Role         string `json:"role"`
	AccessToken  string `json:"access_token"`
	RefreshToken string `json:"refresh_token"`
}

type JWTPayloadClaims struct {
	ID      int64  `json:"user_id"`
	Email   string `json:"email"`
	Role    string `json:"role"`
	Expires int64  `json:"expires"`
	jwt.RegisteredClaims
}


type PersonnelAccounts struct {
	ID       int64  `json:"user_id"`
	FullName string `json:"fullname"`
	Address  string `json:"address"`
	Email    string `json:"email"`
	Role     string `json:"role"`
}
